packages/core/src/tools/create-lambda-handler.js

```diff
@@ -189,7 +189,9 @@ const createLambdaHandler = (appRawOrPath) => {
     // Wait for all async events to complete before callback returns.
     // This is not strictly necessary since this is the default now when
     // using the callback; just putting it here to be explicit.
-    context.callbackWaitsForEmptyEventLoop = true;
+    // In some cases, the code hangs and never exits because the event loop is not
+    // empty, so we can override the default behavior and exit after the app is done.
+    context.callbackWaitsForEmptyEventLoop = !event.skipWaitForAsync || true;
```

Comment:
@standielpls I think this is always `true`, no? `false || true === true`, `undefined || true === true`.

```suggestion
    context.callbackWaitsForEmptyEventLoop =  true;
    if (event.skipWaitForAsync === false) {
        context.callbackWaitsForEmptyEventLoop = false;
     }
```

Reply:
Alternatively: `event.skipWaitForAsync !== true;`

Reply:
Whoops, yes you're right!